storagemarket/deal_execution.go

```diff
@@ -498,11 +502,18 @@ func (p *Provider) indexAndAnnounce(ctx context.Context, pub event.Emitter, deal
 	p.dealLogger.Infow(deal.DealUuid, "deal successfully added to piecestore")
 
 	// register with dagstore
-	if err := stores.RegisterShardSync(ctx, p.dagst, pc, deal.InboundFilePath, true); err != nil {
+	if err := stores.RegisterShardSync(ctx, p.dagst, pc, deal.InboundFilePath, true); err != nil && !xerrors.Is(err, dagstore.ErrShardExists) {
 		return fmt.Errorf("failed to register deal with dagstore: %w", err)
 	}
 	p.dealLogger.Infow(deal.DealUuid, "deal successfully registered in dagstore")
 
+	// announce to the network indexer
+	annCid, err := p.ip.AnnounceBoostDeal(ctx, deal)
+	if err != nil && !xerrors.Is(err, provider.ErrAlreadyAdvertised) {
+		return fmt.Errorf("failed to announce deal to index provider: %w", err)
+	}
```

Comment:
Shouldn't we check for err != nil (i.e. when the err is provider.ErrAlreadyAdvertised), before saying that deal is successfully announced?

Reply:
Hey Anton,  if the error is `provider.ErrAlreadyAdvertised`, we have nothing to worry about and can consider announcement to be successful given how the index provider stuff works.

Reply:
Returning both a result and an error is certainly not idiomatic behaviour and somewhat confusing - this is why I mentioned the comment. We basically end up with an error and with the result (`annCid`), which is really not what one expects unless they know the full stack.

Reply:
I see what you mean. Have fixed this.